beacon-chain/powchain/engine_client.go

```diff
@@ -254,13 +254,23 @@ func (s *Service) GetTerminalBlockHash(ctx context.Context) ([]byte, bool, error
 		if parentBlk == nil {
 			return nil, false, errors.New("parent execution block is nil")
 		}
+
 		if blockReachedTTD {
 			parentTotalDifficulty, err := tDStringToUint256(parentBlk.TotalDifficulty)
 			if err != nil {
 				return nil, false, errors.Wrap(err, "could not convert total difficulty to uint256")
 			}
+
+			// If terminal block has time same timestamp or greater than transition time,
+			// then the node violates the invariant that a block's timestamp must be
+			// greater than its parent's timestamp. Execution layer will reject
+			// a fcu call with such payload attributes. It's best that we return `None` in this a case.
 			parentReachedTTD := parentTotalDifficulty.Cmp(terminalTotalDifficulty) >= 0
 			if !parentReachedTTD {
+				if blk.Time >= transitionTime {
+					return nil, false, nil
+				}
+
```

Comment:
review

Reply:
I am worried about this solution, the transition time being passed to this function is purely on the CL side (you're passing a slot time) while the condition we want to prevent is that the block timestamp is not that of the TTD. Why not simply comparing the timestamp of these two blocks here instead of comparing to `transitionTime`?